lighthouse-core/audits/image-size-responsive.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * @license Copyright 2020 Sebastian Kreft All Rights Reserved.
```

Comment:
I've never really paid attention to the license header in new files by non-google employees but the CLA does pretty clearly say that it's just the contributor granting Google an irrevocable license and not a transfer of intellectual property rights like an ordinary employment agreement might require. Interesting 🤔

Reply:
> the CLA does pretty clearly say that it's just the contributor granting Google an irrevocable license and not a transfer of intellectual property rights like an ordinary employment agreement might require

that's true, but as a Google project, Lighthouse is going to be redistributing it under the project's Apache 2.0 license from Google, so we do need the `@license Copyright 2020 Google Inc. All Rights Reserved.`

https://opensource.google/docs/releasing/contributions/#copyright has a bit more

Reply:
@brendankenny That same page says:

> However for projects that accept a great deal of external code (such as Chromium), you may wish to optionally update the copyright statements in your source code to “Copyright 2014 The [Project] Authors” to acknowledge the community’s efforts. You may do this if you create and maintain AUTHORS and CONTRIBUTORS files. This requires ongoing maintenance but builds goodwill among your contributors.

Which is also in line with t\what the Linux foundation says. See https://www.linuxfoundation.org/blog/2020/01/copyright-notices-in-open-source-software-projects/.

I see the `AUTHORS` file is already there, should I add me there and change the line to

```
@license Copyright 2020 The Lighthouse Authors All Rights Reserved.
```

Reply:
> Lighthouse is going to be redistributing it under the project's Apache 2.0 license from Google

The Apache 2.0 license is the Apache 2.0 license whether from @sk- or Google, so I would've preferred to take another lunch of spirited discussion with the OSS folks to fight this as a "need" if I were still able to do so. Pre-existing licensing knowledge aside, the fact that in the very same guidelines Google happily distributes files in a `third_party` directory with alternative license headers and that the entire "Google Inc." can freely be replaced with "The Project Authors" with a reference to a list real authors once the contributions are significant enough is substantive evidence that this is obviously not a copyright necessity. If anything, claiming that this is the requirement and attempting to adhere to it is actually *detrimental* as it sets the standard of care much higher than it need to be.

If Google would like to require that projects enforce this as a useful delineation between code contributed under a valid CLA to which they have unrestricted copyrights and code relying on a more limited or alternative licensing arrangement, that's all well and good, but claiming it as a legal requirement is FUD and counterproductive. `</rant>`

Reply:
> @brendankenny That same page says:
> 
> > However for projects that accept a great deal of external code (such as Chromium), you may wish to optionally update the copyright statements in your source code to “Copyright 2014 The [Project] Authors” to acknowledge the community’s efforts. You may do this if you create and maintain AUTHORS and CONTRIBUTORS files. This requires ongoing maintenance but builds goodwill among your contributors.
> 
> Which is also in line with what the Linux foundation says

Yep, and it might be a good idea to do that to better celebrate contributions beyond our release notes and the github `contributors` UI. We can open an issue for that and discuss there, but I would imagine it should be done in a single change and not piecemeal. I'd imagine we'd also want to retroactively add (at least some) folks to AUTHORS, but not sure how to handle that.

> If Google would like to require that projects enforce this as a useful delineation between code contributed under a valid CLA to which they have unrestricted copyrights and code relying on a more limited or alternative licensing arrangement, that's all well and good, but claiming it as a legal requirement is FUD and counterproductive.

The only legal claim on there is that signing the CLA and issuing the code under `Copyright Google LLC` doesn't affect the author's rights. I interpret the requirement in that text being on me, as a Google employee, to only accept code contributions of this form. It seems like the delineation you mention *is* the obvious reasoning in that case, but I'm not a lawyer and I don't speak for Google.

(unless you're referring to my comment above, in which case I was speaking project policy, and apologies if it came off as if I were making a legal statement)

Reply:
> We can open an issue for that and discuss there, but I would imagine it should be done in a single change and not piecemeal. I'd imagine we'd also want to retroactively add (at least some) folks to AUTHORS, but not sure how to handle that.

+1 that sounds great! 👍 

And yeah sorry I shot you, the messenger, there @brendankenny. I was ranting at the policy for forcing Google employees to do this in a document that is meant to be legal guidance on OSS when it's a (mostly) separate concern. I understand your unfortunate responsibility to enforce it :)

Reply:
Thanks @patrickhulce for PR #10469.

I updated the license to read "The Lighthouse Authors" and added myself to the `AUTHORS` file.